Src/AutoFixture/NumericSequenceGenerator.cs

```diff
@@ -31,34 +32,18 @@ public object Create(object request, ISpecimenContext context)
 
         private object CreateNumericSpecimen(Type request)
         {
-            var typeCode = Type.GetTypeCode(request);
-
-            switch (typeCode)
+            if(!request.IsNumeric())
+            {
+                return new NoSpecimen();
+            }
+            var random = GetNextNumber();
+            try
+            {
+                return Convert.ChangeType(random, request, CultureInfo.InvariantCulture);
+            }
+            catch(OverflowException)
             {
-                case TypeCode.Byte:
-                    return (byte)this.GetNextNumber();
-                case TypeCode.Decimal:
-                    return (decimal)this.GetNextNumber();
-                case TypeCode.Double:
-                    return (double)this.GetNextNumber();
-                case TypeCode.Int16:
-                    return (short)this.GetNextNumber();
-                case TypeCode.Int32:
-                    return (int)this.GetNextNumber();
-                case TypeCode.Int64:
-                    return this.GetNextNumber();
-                case TypeCode.SByte:
-                    return (sbyte)this.GetNextNumber();
-                case TypeCode.Single:
-                    return (float)this.GetNextNumber();
-                case TypeCode.UInt16:
-                    return (ushort)this.GetNextNumber();
-                case TypeCode.UInt32:
-                    return (uint)this.GetNextNumber();
-                case TypeCode.UInt64:
-                    return (ulong)this.GetNextNumber();
-                default:
-                    return new NoSpecimen();
+                return Convert.ChangeType(0, request, CultureInfo.InvariantCulture);

```

Comment:
I do realise that it was my suggestion to investigate whether `Convert.ChangeType` is an option, but this implementation changes the behaviour of the algorithm.

When I suggested looking into using `Convert.ChangeType`, I wasn't aware that it'd throw an exception if the input value was outside the acceptable range for that type.

That's not what happened before, because the explicit cast would simply overflow the value. As an example, for `byte` values, that means that when we've exhausted [0..255], the algorithm simply loops back and starts over. As an example, `(byte)257` returns `1`. This is exactly the behaviour we want.

With the proposed `try/catch` behaviour, what instead happens is that once the maximum for a type is reached, all further numbers are all `0`.

This FSI session demonstrates this:

```F#
> let nsg = NumericSequenceGenerator ();;
val nsg : NumericSequenceGenerator

> [1..300] |> List.map (fun _ -> nsg.Create(typeof<byte>, null)) |> List.skip 250;;
val it : obj list =
  [251uy; 252uy; 253uy; 254uy; 255uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy;
   0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy;
   0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy;
   0uy; 0uy; 0uy; 0uy; 0uy; 0uy; 0uy]
```

This isn't the behaviour we want. Perhaps we should have a test case that covers that scenario.